llvm/include/llvm/IR/InstrTypes.h

```diff
@@ -60,6 +60,8 @@ class UnaryInstruction : public Instruction {
       : Instruction(Ty, iType, &Op<0>(), 1, IB) {
     Op<0>() = V;
   }
+  LLVM_DEPRECATED("Use BasicBlock::iterators for insertion instead",
+                  "BasicBlock::iterator")
```

Comment:
I think these (and the other cases in this file where you've added `LLVM_DEPRECATED`) could be replaced with `InsertPosition` arguments - especially the `HANDLE_` cases, where we have 3 different methods for each type, which all forward to the same `InsertPosition` fn underneath. Not sure why I missed these the first time, but (as long as it works) it should probably be done now.